### PR TITLE
BUG: GH12071 .reset_index() should create a RangeIndex

### LINKED ISSUE #12071: BUG: .reset_index() should create a RangeIndex
```
In [10]: df = DataFrame({'A' : range(5)})

In [11]: df
Out[11]: 
   A
0  0
1  1
2  2
3  3
4  4

In [12]: df.index
Out[12]: RangeIndex(start=0, stop=5, step=1)

In [13]: df.reset_index()
Out[13]: 
   index  A
0      0  0
1      1  1
2      2  2
3      3  3
4      4  4

In [14]: df.reset_index().index
Out[14]: Int64Index([0, 1, 2, 3, 4], dtype='int64')
```

### PR DESCRIPTION
**Pull request by boombard**
closes #12071

### CODE REVIEW

**Comment by boombard** [comment]
Fixes #12071

File: pandas/core/frame.py [hunk 1]
```diff
@@ -2891,7 +2891,10 @@ def _maybe_casted_values(index, labels=None):
                                                                 np.nan)
             return values
 
-        new_index = np.arange(len(new_obj), dtype='int64')
```
Comment (jreback) [inline pandas/core/frame.py]: just use ``core.common._default_index`` instead here

File: pandas/tests/frame/test_alter_axes.py [hunk 2]
```diff
@@ -578,6 +578,12 @@ def test_reset_index_with_datetimeindex_cols(self):
                                          datetime(2013, 1, 2)])
         assert_frame_equal(result, expected)
 
+    def test_reset_index_range(self):
+        # GH 12071
+        df = pd.DataFrame({'A': range(5)})
+        new_index = df.reset_index().index
+        tm.assertIsInstance(new_index, RangeIndex)
```
Comment (jreback) [inline pandas/tests/frame/test_alter_axes.py]: construct the expected ``DataFrame`` and use ``assert_frame_equal``

**Comment by jreback** [comment]
add this issue number to right after #939 in ``whatsnew/v0.18.0``

**Comment by boombard** [comment]
Thanks. Updated the pull request. Not sure if I put the reference to the issue in the right place in "whatsnew" (this is my first PR to pandas).

File: pandas/core/frame.py (hunk 1 above)
Comment (jreback) [inline pandas/core/frame.py]: this can always be _default_index

Comment (boombard) [inline pandas/core/frame.py]: I changed it and squashed it into the same commit

**Comment by jreback** [comment]
add a test for Series as well

File: pandas/tests/frame/test_alter_axes.py [hunk 3]
```diff
@@ -578,6 +578,24 @@ def test_reset_index_with_datetimeindex_cols(self):
...
+        df = pd.DataFrame([[0, 0], [1, 1]], columns=['A', 'B'],
+                          index=RangeIndex(stop=2))
+        result = df.reset_index()
+        tm.assertIsInstance(result.index, RangeIndex)
+        expected = pd.DataFrame([[0, 0, 0], [1, 1, 1]],
+                                columns=['index', 'A', 'B'],
+                                index=RangeIndex(stop=2))
+        assert_frame_equal(result, expected)
+
+        s = pd.Series(range(2), name='A', index=RangeIndex(stop=2))
```
Comment (boombard) [inline pandas/tests/frame/test_alter_axes.py]: Test for Series Added

File: pandas/core/frame.py (hunk 1 above)
Comment (jreback) [inline pandas/core/frame.py]: ``new_index = _default_index(len(new_obj))``
is the entire answer here, you don't need a separate check whether its a ``RangeIndex`` or not

File: pandas/tests/frame/test_alter_axes.py [hunk 4]
```diff
@@ -578,6 +578,24 @@ def test_reset_index_with_datetimeindex_cols(self):
...
+        series_result = s.reset_index()
```
Comment (jreback) [inline pandas/tests/frame/test_alter_axes.py]: put the series test in ``test_series`` where appropraite

File: pandas/core/frame.py (hunk 1 above)
Comment (boombard) [inline pandas/core/frame.py]: Fixed

**Comment by jreback** [comment]
merged via https://github.com/pydata/pandas/commit/16a7799702f43d58ee2c2b3582c5a2c92b34fc0c

thanks!

**Comment by boombard** [comment]
Thanks for the help @jreback

**Pull request closed without merging**